build_tools/compiler/template_processor.rb

```diff
@@ -39,5 +40,71 @@ def asset_path(file, options={})
     def method_missing(name, *args)
       puts "#{name} #{args.inspect}"
     end
+
+    def stylesheet_link_tag(*sources)
+      options = stringify_keys(extract_options!(sources))
+      sources.uniq.map { |source|
+        link_options = {
+            "rel" => "stylesheet",
+            "media" => "screen",
+            "href" => asset_path(source)
+        }.merge!(options)
+        tag(:link, tag_options(link_options))
+      }.join("\n")
+    end
+
+    def javascript_include_tag(*sources)
+      options = stringify_keys(extract_options!(sources))
+      sources.uniq.map { |source|
+        script_options = {
+            "src" => asset_path(source)
+        }.merge!(options)
+        content_tag(:script, tag_options(script_options))
+      }.join("\n")
+    end
+
+    def content_tag(name, options = nil)
+      "<#{name}#{options}></#{name}>"
+    end
+
+    def tag(name, options)
+      "<#{name}#{options}/>"
+    end
+
+    def extract_options!(options)
```

Comment:
If we added a require for `activesupport/core/ext/array` as well as the hash one then we could also rely on `Array#extract_options!` and not need to implement it here.  It's also in rails 3.1 and wouldn't break our dependency (see: http://guides.rubyonrails.org/v3.1/active_support_core_extensions.html#options-extraction)